util/align_ptr.h

```diff
@@ -0,0 +1,32 @@
+#ifndef _ALIGN_PTR_H_
+#define _ALIGN_PTR_H_
+
+/**
+ * ALIGN_PTR_DECL(type, name, num):
+ * Declare a variable array called ${name}, large enough to contain an array
+ * of ${num} values of ${type}.  Must not be used until ALIGN_PTR_INIT() has
+ * been called.  This macro may also create an additional variable called
+ * "${name}_buf".
+ */
+#define ALIGN_PTR_DECL(type, name, num)					\
```

Comment:
Hmm, I just realized that this won't help in `crypto_aesctr.c`.  In that case, we want a `uint8_t` array that's aligned to satisfy `__m128i`.

I'll mess around a bit more, and probably add a REBASE commit to this PR.